src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -38,20 +44,12 @@ export class LiteralExpressionEvaluator extends ASTMapper {
     // It is sometimes possible to avoid any calculation and take the value from the type
     // This is not always possible because boolean literals do not contain their value in the type,
     // and because very large int and rational literals omit some digits
-    const result = createLiteralFromType(node.typeString) || evaluateLiteralExpression(node);
-
-    if (result === null) {
-      this.commonVisit(node, ast);
-    } else if (typeof result === 'boolean') {
-      ast.replaceNode(node, createBoolLiteral(result, ast));
-    } else {
-      const intValue = result.toInteger();
-      if (intValue === null) {
-        throw new TranspileFailedError('Attempted to make node for non-integral literal');
-      }
-
-      ast.replaceNode(node, createNumberLiteral(intValue, ast));
+    let result = createLiteralFromType(node, ast);
```

Comment:
It should be verified with infertype class - handling of types there is different. For example literal strings don't contain string value in typestring anymore afaik

Reply:
I've made an [issue](https://github.com/NethermindEth/warp/issues/971) for that

Reply:
Thanks, I had just used the implementation we had in this pass for this method, and made the necessary changes to comply to the new return type of the function; but I didn't notice this had to be modified for the solc-typed-ast update. Good catch.